src/ZLogger.Unity/Assets/Scripts/ZLogger/AsyncStreamLineMessageWriter.cs

```diff
@@ -1,62 +1,212 @@
 using System;
+using System.Buffers;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Runtime.CompilerServices;
 using System.Text;
 using System.Threading;
 using System.Threading.Channels;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using ZLogger.Entries;
 
 namespace ZLogger
 {
     public class AsyncStreamLineMessageWriter : IAsyncLogProcessor, IAsyncDisposable
     {
         readonly byte[] newLine;
-        readonly bool crlf;
-        readonly byte newLine1;
-        readonly byte newLine2;
-
-        readonly Stream stream;
-        readonly Channel<IZLoggerEntry> channel;
-        readonly Task writeLoop;
-        readonly ZLoggerOptions options;
+        readonly bool   crlf;
+        readonly byte   newLine1;
+        readonly byte   newLine2;
+
+        readonly Stream                  stream;
+        readonly Channel<IZLoggerEntry>  channel;
+        readonly Task                    writeLoop;
+        readonly Task                    summaryWriteLoop;
+        readonly ZLoggerOptions          options;
         readonly CancellationTokenSource cancellationTokenSource;
 
         public AsyncStreamLineMessageWriter(Stream stream, ZLoggerOptions options)
         {
-            this.newLine = Encoding.UTF8.GetBytes(Environment.NewLine);
+            this.newLine                 = Encoding.UTF8.GetBytes(Environment.NewLine);
             this.cancellationTokenSource = new CancellationTokenSource();
             if (newLine.Length == 1)
             {
                 // cr or lf
                 this.newLine1 = newLine[0];
                 this.newLine2 = default;
-                this.crlf = false;
+                this.crlf     = false;
             }
             else
             {
                 // crlf(windows)
                 this.newLine1 = newLine[0];
                 this.newLine2 = newLine[1];
-                this.crlf = true;
+                this.crlf     = true;
             }
 
             this.options = options;
-            this.stream = stream;
+            this.stream  = stream;
             this.channel = Channel.CreateUnbounded<IZLoggerEntry>(new UnboundedChannelOptions
             {
                 AllowSynchronousContinuations = false, // always should be in async loop.
-                SingleWriter = false,
-                SingleReader = true,
+                SingleWriter                  = false,
+                SingleReader                  = true,
             });
-            this.writeLoop = Task.Run(WriteLoop);
+
+            this.writeLoop        = Task.Run(WriteLoop);
+            this.summaryWriteLoop = Task.Run(SummaryWriteLoop);
         }
 
+        private const bool ENABLE_SPAM_DROPPER  = true;
+        private const bool DEBUG_SPAM           = false;
+        private const uint LIMIT_IN             = 3;
```

Comment:
we need to fix those limits before merging

Reply:
I Suggest:

50 / 1sec In
40 / 1sec Out (stop dropping logs after spam rate reaches the about condition)

Reply:
the thresholds should be bigger to accommodate for spikes in log writes (scene load etc.).
and we need to assume 45fps.
so maybe 200/5sec instead? (over 40 messages for 5 sec)

also now that I'm thinking about it, we need a way to disable it in editor.

can you test our production code (not a sample scene) with these scenarios?

1. combat, a script throws an error every frame
2. combat, 3 scripts throw an error every frame
3. main menu, a script throws an error every frame